src/cljam/util/chromosome.clj

```diff
@@ -48,3 +49,10 @@
   [s]
   (some? (re-matches #"^chr([0-9]{1,2}|X|Y|M|MT)"
                      (normalize-chromosome-key s))))
+
+(defn chromosome-order-key [s]
+  (if-let [[_ _ chr suffix] (re-find #"(?i)^(chr)?([1-9][0-9]*|X|Y|M|MT)(\S*)" s)]
```

Comment:
`MT` is never trapped by this regex.

```clojure
(chromosome-order-key "chrMT_foo")
;;=> [2147483645 "T_foo"]
```

This causes an ordering error in some cases. For example,

```clojure
(sort-by chromosome-order-key ["chrM_foo" "chrMT_foo"])
;;=> ("chrMT_foo" "chrM_foo")
```